src/Common/cmake/secure_dependencies.cmake

```
@@ -0,0 +1,73 @@
+#===============================================================================
+# Configure the flags used to ensure that required security settings are enabled
+macro(configure_security_flags)
+    # Enable Control Flow Guard
+    if (WIN32)
+        add_link_options("LINKER:/guard:cf")
+    endif()
+
+    # Enable Compiler supported Spectre mitigations
+    if (NOT APPLE)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mspeculative-load-hardening -mretpoline")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mspeculative-load-hardening -mretpoline")
+    endif()
+endmacro()
+
+#===============================================================================
+# Set MSVC static runtime library policy
+# NOTE: Since this sets a cmake policy for ensuring Windows compilation uses static runtime,
+# this policy MUST be set before any projects are declared for it to take effect.
+macro(set_msvc_static_runtime_policy)
+    # Enforce use of static runtime (avoids target machine needing msvcrt installed).
+    # Must set policy CMP0091 before any projects are declared (see https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html#cmake-msvc-runtime-library)
+    cmake_policy(SET CMP0091 NEW)
+    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+endmacro()
```

The fragment "for ensuring **Windows** compilation uses static runtime" makes me think that the `if (WIN32)` might be needed:
```suggestion
macro(set_msvc_static_runtime_policy)
    if (WIN32)
        # Enforce use of static runtime (avoids target machine needing msvcrt installed).
        # Must set policy CMP0091 before any projects are declared (see https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html#cmake-msvc-runtime-library)
        cmake_policy(SET CMP0091 NEW)
        set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
    endif (WIN32)
endmacro()
```

---

The CMake API docs clarify that this value is ignored on non-Windows platforms so it is safe to always set:
```
The value is ignored on non-MSVC compilers but an unsupported value will be rejected as an error when using a compiler targeting the MSVC ABI.
```

---

(I assume that with "The value is ignored" you mean "the  CMAKE_MSVC_RUNTIME_LIBRARY is ignored")

You write: "this value is ignored **on non-Windows platforms**".
The citation writes: "The value is ignored **on non-MSVC compilers**".
Our compiler is **clang**/**clang++** (including on Win), I assume it is non-MSVC compiler. Which makes me think that the value is ignored even on Windows.
This all sounds suspicious ;-), but unfortunately I cannot (at least quickly) provide any sensible recommendation. Leaving it to you.

---

Disregard, this is existing/old code.

---

You are right, the wording in the docs is misleading. The page I'm looking at is the docs for [CMAKE_MSVC_RUNTIME_LIBRARY](https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html#cmake-msvc-runtime-library), which at the top says:

> Select the MSVC runtime library for use by compilers targeting the MSVC ABI. This variable is used to initialize the [MSVC_RUNTIME_LIBRARY](https://cmake.org/cmake/help/latest/prop_tgt/MSVC_RUNTIME_LIBRARY.html#prop_tgt:MSVC_RUNTIME_LIBRARY) property on all targets as they are created.

In this case, while we are using clang as our compiler instead of MSVC, we are using clang to compile a binary that targets the MSVC runtime, which means CMake needs to use logic to decide which runtime to pass to the compilation and linking. So even though we call into clang and lld-link, the latter is the LLVM linker for Windows platforms that targets MSVC and as such obeys these CMake variables (for more info check out https://lld.llvm.org/windows_support.html).